Makefile

```diff
@@ -45,7 +45,7 @@ INSTALL=install
 # checkout. Invoke make with GOOGLE_PROTOBUF_CHECKOUT=[PATH_TO_CHECKOUT] to
 # override this value, i.e. -
 #   make [TARGET] GOOGLE_PROTOBUF_CHECKOUT=[PATH_TO_CHECKOUT]
-GOOGLE_PROTOBUF_CHECKOUT=../protobuf
+GOOGLE_PROTOBUF_CHECKOUT?=../protobuf
```

Comment:
We don't actually document using the environment for this, the info right above only mentions passing it via `make`, if we want to support the env for these, shouldn't we do it for others?  (`PROTOC`, etc.)